### PR TITLE
make qphix_test_Dslash meaningful

### PR DESCRIPTION
**Pull request by kostrzewa**
* provide support for reading operator type from input file
* implement single application of full operator in qphix interface

### CODE REVIEW

**Comment by kostrzewa** [comment]
Okay, so I think we're ready to finish working on the clover stuff now!

File: qphix_interface.cpp [hunk 1]
```diff
@@ -819,8 +814,10 @@ int invert_eo_qphix_helper(spinor *const tmlqcd_even_out, spinor *const tmlqcd_o
   double end = omp_get_wtime();
 
   uint64_t num_cb_sites = lattSize[0] / 2 * lattSize[1] * lattSize[2] * lattSize[3];
+  // FIXME: this probably needs to be adjusted depending on the operator used
   uint64_t total_flops = (site_flops + (72 + 2 * 1320) * mv_apps) * num_cb_sites;
   QPhiX::masterPrintf("# Solver Time = %g sec\n", (end - start));
+  QPhiX::masterPrintf("# Flops per site = %ld \n", site_flops );
```
Comment (kostrzewa) [inline qphix_interface.cpp]: I should probably remove this, I don't understand its value...

Comment (martin-ueding) [inline qphix_interface.cpp]: You mean the `total_flops`? In the QPhiX tests, `site_flops` is used to track the number of operations from the BLAS routines and `mv_apps` is the number of fermion matrix operator applications.

**Comment by martin-ueding** [comment]
So the interface for Wilson and Wilson twisted mass is done? That sounds great! 😃 

Do we want to continue with Wilson clover or TM clover?

File: qphix_interface.cpp (hunk 1 above)
Comment (kostrzewa) [inline qphix_interface.cpp]: I meant site flops. Okay, but then the constant multiplying mv_apps is not correct. 1320 is for Wilson only.

**Comment by kostrzewa** [comment]
> So the interface for Wilson and Wilson twisted mass is done? That sounds great! :smiley:

Up to the problem with communication in two space-time directions simultaneously... I was planning to look at the QDP packers to understand this.

**Comment by kostrzewa** [comment]
This is pulled in by #383, closing.

**Pull request closed without merging**